src/services/continuedev/core/autocomplete/util/AutocompleteLruCacheInMem.ts

```diff
@@ -0,0 +1,77 @@
+import { LRUCache } from "lru-cache"
+
+const MAX_PREFIX_LENGTH = 50000
+
+function truncatePrefix(input: string, safety: number = 100): string {
+	const maxBytes = MAX_PREFIX_LENGTH - safety
+	let bytes = 0
+	let startIndex = 0
+
+	// Count bytes from the end, keeping the most recent typing
+	for (let i = input.length - 1; i >= 0; i--) {
+		bytes += new TextEncoder().encode(input[i]).length
+		if (bytes > maxBytes) {
+			startIndex = i + 1
+			break
+		}
+	}
+
+	return input.substring(startIndex)
+}
+
+export class AutocompleteLruCacheInMem {
+	private static capacity = 1000
```

Comment:
```suggestion
	private static capacity = 100
```

Given it's in mem, and in weird corner cases this might get big, I vote for keeping this number as low as we think it reasonably can go without significantly impacting hit rate.  I'm just guessing here, but after 101 autocomplete requests it strikes me as very low probability that there's a great hit rate on the old cache entries.